src/Microsoft.NetCore.Analyzers/UnitTests/Performance/UsePropertyInsteadOfCountMethodWhenAvailableTests.cs

```diff
@@ -0,0 +1,452 @@
+}}
+public static class C
+{{
+    public static T GetData() => default;
+    public static int M() => GetData().Count;
+}}
+");
+
+        [Fact]
+        public static Task CSharp_ICollectionOfTImplementerWithExplicitCount_NoDiagnostic()
+            => VerifyCS.VerifyAnalyzerAsync(
+                $@"using System;
+using System.Linq;
+public class T : global::System.Collections.Generic.ICollection<string>
+{{
+    int global::System.Collections.Generic.ICollection<string>.Count => throw new NotImplementedException();
+    bool global::System.Collections.Generic.ICollection<string>.IsReadOnly => throw new NotImplementedException();
+    void global::System.Collections.Generic.ICollection<string>.Add(string item) => throw new NotImplementedException();
+    void global::System.Collections.Generic.ICollection<string>.Clear() => throw new NotImplementedException();
+    bool global::System.Collections.Generic.ICollection<string>.Contains(string item) => throw new NotImplementedException();
+    void global::System.Collections.Generic.ICollection<string>.CopyTo(string[] array, int arrayIndex) => throw new NotImplementedException();
+    public global::System.Collections.Generic.IEnumerator<string> GetEnumerator() => throw new NotImplementedException();
+    global::System.Collections.IEnumerator global::System.Collections.IEnumerable.GetEnumerator() => throw new NotImplementedException();
+    bool global::System.Collections.Generic.ICollection<string>.Remove(string item) => throw new NotImplementedException();
+}}
+public static class C
+{{
+    public static T GetData() => default;
+    public static int M() => GetData().Count();
+}}
+");
+
+        [Fact]
+        public static Task CSharp_InterfaceShadowingICollectionOfT_NoDiagnostic()
+            => VerifyCS.VerifyAnalyzerAsync(
+                @"using System;
+using System.Linq;
+public interface I : global::System.Collections.Generic.ICollection<string>
+{
+    new int Count { get; }
+}
+public static class C
+{
+    public static I GetData() => default;
+    public static int M() => GetData().Count();
+}
+");
+
+        [Fact]
+        public static Task CSharp_InterfaceShadowingICollection_NoDiagnostic()
+            => VerifyCS.VerifyAnalyzerAsync(
+                @"using System;
+using System.Linq;
+public interface I :
+    global::System.Collections.Generic.IEnumerable<string>,
+    global::System.Collections.ICollection
+{
+    new int Count { get; }
+}
+public static class C
+{
+    public static I GetData() => default;
+    public static int M() => GetData().Count();
+}
+");
+
+        [Fact]
+        public static Task CSharp_ClassShadowingICollectionOfT_NoDiagnostic()
+            => VerifyCS.VerifyAnalyzerAsync(
+                $@"using System;
+using System.Linq;
+public class T : global::System.Collections.Generic.ICollection<string>
+{{
+    public int Count => throw new NotImplementedException();
```

Comment:
Shouldn't we be flagging this case? Looking at the added unit tests here, I wouldn't mind us loosening the `GetReplacementProperty` in the analyzer to just check for the presence of a public int property named `Length` or `Count`. Personally, I don't think that will lead to any _real_ false positives, and would benefit cases such as this one being flagged.

Reply:
Simplified the logic but, do to the substancial differences between C# and Visual Basic, I think it's better to repeat some code and avoid more virtual method invocations.

Reply:
Maybe I'm too restrictive, but I'm making sure the only the interface implementations are flagged.